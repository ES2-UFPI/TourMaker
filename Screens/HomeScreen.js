import React, { Component } from 'react';
import { View } from 'react-native';

import CustomMapView from './ScreensModules/CustomMapView';
import CustomButton from './ScreensModules/CustomButton';

import Styles from './Styles';

class HomeScreen extends Component {

    render() {
        return (
            <View style={Styles.Screen}>
                <CustomMapView
                    EnableNearbySeach={true}
                />
                <CustomButton
                    title="Criar Roteiro Automatico"
                    color={Styles.NavigationButtons.color}
                    onPress={() => {
                        this.props.navigation.push('RoteiroAutomatico', {})
                    }}
                />
                <CustomButton
                    title="Criar Roteiro Manual"
                    color={Styles.NavigationButtons.color}
                    onPress={() => {
                        this.props.navigation.push('RoteiroManual', {})
                    }}
                />
                <CustomButton
                    title="Roteiros Salvos"
                    color={Styles.NavigationButtons.color}
                    onPress={() => {
                        this.props.navigation.push('SavedRoutes', {})
                    }}
                />
            </View>
        )
    }
}

export default HomeScreen;