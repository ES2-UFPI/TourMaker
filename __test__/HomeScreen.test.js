import React from 'react';
import renderer from 'react-test-renderer';

import HomeScreen from '../Screens/HomeScreen';

describe('<HomeScreen />', () => {
    it('has 3 child', () => {
        const tree = renderer.create(<HomeScreen />).toJSON();
        expect(tree.children.length).toBe(4);
    });
});

it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});