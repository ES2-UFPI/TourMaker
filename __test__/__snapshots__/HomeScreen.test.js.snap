// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "yellow",
        "height": 400,
        "justifyContent": "flex-start",
        "width": "100%",
      }
    }
  >
    <AIRMap
      followsUserLocation={false}
      handlePanDrag={false}
      initialRegion={null}
      onChange={[Function]}
      onLayout={[Function]}
      onMapReady={[Function]}
      onMarkerPress={[Function]}
      onRegionChangeComplete={[Function]}
      onUserLocationChange={[Function]}
      region={null}
      showsUserLocation={false}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#1abc9c",
                },
              ]
            }
          >
            Criar Roteiro Automatico
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#1abc9c",
                },
              ]
            }
          >
            Criar Roteiro Manual
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#1abc9c",
                },
              ]
            }
          >
            Roteiros Salvos
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
